starknet_py/net/networks.py

```diff
@@ -11,7 +11,7 @@
 except ImportError:
     PredefinedNetwork = str
 
-Network = Union[PredefinedNetwork, str]
+Network = Union[PredefinedNetwork, str, dict]
```

Comment:
Maybe we could use TypedDict here so it's more explicit what keys we expect this dict to have?